src/django_components/component_registry.py

```diff
@@ -66,10 +77,16 @@ class ComponentRegistry:
     ```
     """
 
-    def __init__(self, library: Optional[Library] = None) -> None:
+    def __init__(
+        self,
+        library: Optional[Library] = None,
+        settings: Optional[Union[RegistrySettings, Callable[["ComponentRegistry"], RegistrySettings]]] = None,
```

Comment:
I allowed to define the settings also as a getter, so the latest value can be accessed.